modules/3d/include/opencv2/3d.hpp

```diff
@@ -2684,47 +2684,58 @@ class CV_EXPORTS Octree {
     Octree();
 
     /** @overload
-    * @brief Create an empty Octree and set the maximum depth.
-    *
-    * @param maxDepth The max depth of the Octree. The maxDepth > -1.
-    */
-    explicit Octree(int maxDepth);
+     * @brief Creates an empty Octree with given maximum depth
+     *
+     * @param maxDepth The max depth of the Octree
+     * @param size bounding box size for the Octree
+     * @param origin Initial center coordinate
+     * @param withColors Whether to keep per-point colors or not
+     * @return resulting Octree
+     */
+    static Octree createWithDepth(int maxDepth, double size, const Point3f& origin = { }, bool withColors = false);
 
     /** @overload
-    * @brief Create an Octree from the PointCloud data with the specific max depth.
-    *
-    * @param pointCloud Point cloud data.
-    * @param maxDepth The max depth of the Octree.
-    */
-    Octree(const std::vector<Point3f> &pointCloud, int maxDepth);
+     * @brief Create an Octree from the PointCloud data with the specific maxDepth
+     *
+     * @param maxDepth Max depth of the octree
+     * @param pointCloud point cloud data, should be 3-channel float array
+     * @param colors color attribute of point cloud in the same 3-channel float format
+     * @return resulting Octree
+     */
+    static Octree createWithDepth(int maxDepth, InputArray pointCloud, InputArray colors = noArray());
```

Comment:
Input arrays usually go first in the public API.

Reply:
This was made intentionally to reduce amount of overloaded methods